File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
         var vowels = Set.of('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U');
        int a = 0, b = 0;
        for (int i = 0, j = s.length() - 1; i < j; ++i, --j) {
            a += vowels.contains(s.charAt(i)) ? 1 : 0;
            b += vowels.contains(s.charAt(j)) ? 1 : 0;
        }
        return a == b;
    }
}

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution {
public:
    bool halvesAreAlike(string s) {
      int mid=(0+s.size()-1)/2;
    int count1=0, count2=0;
    
    for(int i=0;i<=mid;i++)
    {
        if(s[i]=='a'|| s[i]=='e'|| s[i]== 'i'|| s[i]== 'o'|| s[i]=='u'|| s[i]=='A'|| s[i]=='E'|| s[i]=='I'|| s[i]=='O'|| s[i]=='U')
            count1++;
    }
    
    for(int i=mid+1;i<s.size();i++)
    {
        if(s[i]=='a'|| s[i]=='e'||s[i]== 'i'||s[i]== 'o'||s[i]== 'u'||s[i]== 'A'||s[i]== 'E'||s[i]== 'I'||s[i]== 'O'|| s[i]=='U')
            count2++;
    }
    
    if(count1==count2)
        return true;
    else return false;
    }
};